packages/components/src/components/domainproperties/dataset/DatasetColumnMappingPanel.tsx

```diff
@@ -0,0 +1,163 @@
+
+        this.props.onColumnMappingChange(this.state.closestParticipantIdField, this.state.closestTimepointField);
+    }
+
+    compareNames(name1: string, name2: string): boolean {
+        return name1 === name2 || name1.indexOf(name2) >= 0 || name2.indexOf(name1) >= 0;
+    }
+
+    areColumnNamesAndTypesEquivalent(targetColumnName: string, inferredField: DomainField): boolean {
+        const { subjectColumnName, timepointType } = this.props;
+
+        const targetColumn = targetColumnName.toLowerCase().replace(' ', '');
+        const inferredFieldName = inferredField.name.toLowerCase().replace(' ', '');
+
+        if (targetColumnName === subjectColumnName) {
+            return this.compareNames(targetColumn, inferredFieldName) && inferredField.rangeURI === 'xsd:string';
+        } else if (targetColumnName == timepointType && timepointType === 'DATE') {
+            return this.compareNames('visitdate', inferredFieldName) && inferredField.rangeURI === 'xsd:datetime';
+        } else if (targetColumnName == timepointType && timepointType === 'VISIT') {
+            return this.compareNames('sequencenum', inferredFieldName) && inferredField.rangeURI === 'xsd:double';
+        } else if (targetColumnName == timepointType) {
+            return this.compareNames('date', inferredFieldName) && inferredField.rangeURI === 'xsd:datetime';
+        }
+    }
+
+    findClosestColumn(targetColumn: string): string {
+        const inferredColumns = this.props.model.domain.fields;
+
+        const matchedField = inferredColumns.find(field => {
+            let matchingNames = false;
+
+            if (field.name && field.rangeURI) {
+                matchingNames = this.areColumnNamesAndTypesEquivalent(targetColumn, field);
+            }
+            return matchingNames;
+        });
+
+        return matchedField ? matchedField.name : undefined;
+    }
+
+    onSelectChange = (name, formValue, selected): void => {
+        const value = selected ? selected.name : undefined;
+
+        this.setState(
+            () => ({ [name]: value }),
+            () => {
+                this.props.onColumnMappingChange(
+                    this.state.closestParticipantIdField,
+                    this.state.closestTimepointField
+                );
+            }
+        );
+    };
+
+    render() {
+        const { model } = this.props;
+        const { closestParticipantIdField, closestTimepointField } = this.state;
+
+        const domain = model.domain;
+
+        return (
+            <>
+                <SectionHeading title="Column mapping" />
+                <div className="margin-top">
+                    Columns already existing in the domain can be mapped with columns from your file. Choose a column to
+                    match your ParticipantID and Timepoints.
+                </div>
+                <Row className="margin-top">
+                    <Col xs={4}>
+                        <DomainFieldLabel label="ParticipantID" />
```

Comment:
Similar to above, we'll want to have this label based on the study moduleContext subject info.